### PR TITLE
Crucial routine for IQM coupler

### PR DESCRIPTION
**Pull request by maxhant**
This is the equivalent of the chevron but for a chip with couplers. It allows you to find the idle point as well where the interaction is off. 

What it does: 
The user needs to put the frequency of the qubits at the interaction point where the 01 transition of one qubit is resonant with the 12 (for CZ) / 01 (for iSWAP) transitions of the other qubit (higher frequency). 
This flux pulse will turn on the interaction and change the coupling strength.  

Action:
```yaml
  - id: coupler_cz_flux_time
    priority: 0
    operation: coupler_cz_flux_time
    qubits: [(0,1)]
    # main: coupler_swap
    parameters:
      amplitude_factor_min: -0.55
      amplitude_factor_max: -0.45
      amplitude_factor_step: 0.005 # 0.01
      duration_min: 4
      duration_max: 80
      duration_step: 2
      nshots: 1000
      native_gate: "iSWAP"
```

One should measure the following for the CZ:
image

This is the cleaned-up version of one routine in #340. Note that the original routine was including three driver fixes: 
- Driver was crashing for more than 750 points so I was splitting the sweep in the routine 
- Pulse amplitude was divided in the driver which was changing the value in qibocal leading to wrong plotting 
- Multiplex was not working so I was measuring twice.

The old format was tested on hardware, but this one only tested on the dummy.

I will implement the fitting if I am done with the other PRs. But maybe it can be merged without a fitting?

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: src/qibocal/protocols/characterization/couplers/cz_flux_time.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+def sort_frequency(pair: QubitPair):
+    """Sorts the qubits in a pair by frequency."""
+    if pair.qubit1.frequency > pair.qubit2.frequency:
+        return pair.qubit2, pair.qubit1
+    else:
+        return pair.qubit1, pair.qubit2
+
+
+cz_flux_time = Routine(_aquisition, _fit, _plot)
```
[the remark this answers is not in the document]
Comment (Jacfomg) [reply src/qibocal/protocols/characterization/couplers/cz_flux_time.py:265]: Okay I saw the init but still

**Comment by Jacfomg** [comment]
All 3 issues have been solved in #479 from qibolab. Not tested on the hardware, only in the scope and seemed to work.

**Comment by Jacfomg** [comment]
The plotting is a bit weird from now and tests seem to get dummy instead of dummy2

**Comment by andrea-pasquale** [comment]
> The plotting is a bit weird from now and tests seem to get dummy instead of dummy2

The lock file does not seem to updated. This is probably why.

**Comment by Jacfomg** [comment]
It was also locally, I will try to have a look next week.

**Comment by andrea-pasquale** [comment]
Same as #340

**Comment by andrea-pasquale** [comment]
Can we delete this branch @Jacfomg ?

**Pull request closed without merging**